### ISSUE TITLE
Angular Material Tooltip stop working after virtual scrolling

### ISSUE THREAD

**Issue by sheiidan** [swimlane/ngx-datatable#1321]
**I'm submitting a ...**  (check one with "x")
```
[ x] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ ] support request => Please do not submit support request here, post on Stackoverflow or Gitter
```

**Current behavior**
When one of the datatable cells contains an Angular Material2 Tooltip component, it will stop working after the user scrolls (the datatable is configured with vertical virtual scrolling). When this happens, you can see a blank tooltip shows at the top left corner of the page.

**Expected behavior**
Material Tooltip should always work regardless the user scrolls or not

**Reproduction of the problem**
1. Browse https://stackblitz.com/edit/angular-mq8vbn
2. move the cursor over the 1st column cell values, you should see 'Hello World' tooltips
3. now scroll down several rows
4. move the cursor over the 1st column cell values, this time the tooltip won't work.

(I suspect it might be a zone issues)

**What is the motivation / use case for changing the behavior?**
It breaks the tooltip

**Please tell us about your environment:**
N/A

* **Table version:** latest

* **Angular version:** 5.0.0

* **Browser:** [all | Chrome XX | Firefox XX | IE XX | Safari XX | Mobile Chrome XX | Android X.X Web Browser | iOS XX Safari | iOS XX UIWebView | iOS XX WKWebView ]

* **Language:** [all | TypeScript X.X | ES6/7 | ES5]

**Comment by sheiidan** [comment]
Any update on this issue? It also happens when opening a material dialog after virtual scrolling. The dialog won't open unless you wrap it with zone.run()

**Comment by GITHuHo** [comment]
We experience a similar issue with the ng-bootstrap tooltip if used with a template inside a cell. 
The template's data-bindings get lost after scrolling the table rows in and out of view.

Before scrolling...
![github_issue_ngxdatatable_1321_1](https://user-images.githubusercontent.com/9657001/37771157-0bb5b0e0-2dd7-11e8-8771-148cbbe98cb7.png)

After scrolling...
![github_issue_ngxdatatable_1321_2](https://user-images.githubusercontent.com/9657001/37771170-1186007e-2dd7-11e8-82ff-d3dda58a43dc.png)

```
<ng-template let-value="value" ngx-datatable-cell-template>
    <ng-template #valueToolTip>
        <span>The value is {{value}}</span>
    </ng-template>
    <div [ngbTooltip]="valueToolTip">
        {{value}}
    </div>
</ng-template>
```

This issue only occurs,
- if virtualization is enabled in combination with `[scrollV]="true"`,
- if Angular 5 is used (Angular 4 is fine with any version of ngx-datatable, such as 10.3 or 11.2),
- if a template for ngb-tooltip is used within the `ngx-datatable-cell-template`. 
  Directly binding the tooltip text using the `ngbTooltip` input property is fine, e.g.
  `<div [ngbTooltip]="'The value is ' + {{value}} ">`. In general, using templates inside 
  `ngx-datatable-cell-template` is also working, e.g.:
```
<ng-template let-value="value" ngx-datatable-cell-template>
        <ng-template #valueToolTip>
             <span>The value is {{value}}</span>
        </ng-template>
        <div>
             <ng-container *ngTemplateOutlet="valueToolTip"></ng-container>
        </div>
</ng-template>
```

Turning off virtualization (`<ngx-datatable [virtualization]="false">`) or switching to Angular 4 is not an option for us.

**Comment by sheiidan** [comment]
I found the root cause:

in file \src\components\body\**scroller.component.ts** at `ngOnInit()`,
The usage of `this.ngZone.runOutsideAngular` cause the zone issues.
**removing `this.ngZone.runOutsideAngular` solves the issue**.

I think that the reason for using this runOutsideAngular() is for perf reasons, however it breaks functionality - in case virtual scrolling is on, you might expect Zone issues and weird things will happen (tooltip not working properly, dialogs opened weird, etc...)

```ts
  ngOnInit(): void {
    // manual bind so we don't always listen
    if (this.scrollbarV || this.scrollbarH) {
      const renderer = this.renderer;
      this.parentElement = renderer.parentNode(renderer.parentNode(this.element));
      this.ngZone.runOutsideAngular(() => {
          this.parentElement.addEventListener('scroll', this.onScrolled.bind(this));
      });
    }
  }
```

**Comment by sheiidan** [comment]
This regression came in this commit:
amcdnl@https://github.com/swimlane/ngx-datatable/commit/7227e559107f531fdeaf056d9c79133dcb62fdbf

fix(perf): improved scroll performance

**Comment by sheiidan** [comment]
This regression came in this commit:
@amcdnl@7227e55

fix(perf): improved scroll performance

**Comment by marklbarlow** [comment]
This seems to be the root cause for about a dozen zone / change detection-related problems in the issues list.  Really surprising that this hasn't been fixed - the problem has been there since before Christmas and I've had to keep using v11.0.4 since then as a result.

Is this project still actively maintained and will it ever be fixed?  I hope the answer is "yes" to both as it's a really useful datatable, but the way things are going it might be time to switch to something else...

**Comment by yozo1984** [comment]
+1 We are also seeing this error happening when using the datatable. I manually removed the runOutsideAngular wrap which fixed our problem.

**Comment by sheiidan** [comment]
I also removed `runOutsideAngular`, it fixed the problem and I don't see any performance issues without it

**Comment by ryanohearn** [comment]
Routerlinks stop working when i scroll my data table. I can only get the links to work when I change

```
this.ngZone.runOutsideAngular(() => {
          this.parentElement.addEventListener('scroll', this.onScrolled.bind(this));
});
```

to 
```
this.parentElement.addEventListener('scroll', this.onScrolled.bind(this));
```

**Comment by deeg** [comment]
If anyone is interested to use a version with scrolling running in zone, you can replace your package.json:

"@swimlane/ngx-datatable": "semanticresearch/ngx-datatable#scroll-in-zone",

The only difference from release v13 is making scrolling happen inside zone:

https://github.com/semanticresearch/ngx-datatable/commits/scroll-in-zone

**Comment by marjan-georgiev** [comment]
Fixed in 13.0.1
